experiments/bid_vs_cot.py

```
@@ -0,0 +1,342 @@
+            )
+            text = judge_resp.content[0].text
+            start = text.find('{')
+            end = text.rfind('}') + 1
+            if start == -1 or end == 0:
+                raise ValueError("No JSON found")
+            data = json.loads(text[start:end])
+            if not all(k in data for k in ("score", "justification", "violation")):
+                raise ValueError("Missing fields")
+            score = int(data["score"])
+            if not (1 <= score <= 5):
+                raise ValueError("Score out of range")
```

Ruff (0.15.10)

[warning] Abstract `raise` to an inner function

(TRY301)